tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -1740,16 +1740,17 @@ MLIRScanner::EmitGPUCallExpr(clang::CallExpr *expr) {
         builder.create<mlir::NVVM::Barrier0Op>(loc);
         return make_pair(ValueCategory(), true);
       }
-      if (sr->getDecl()->getIdentifier() &&
+      if (sr->getDecl()->getIdentifier() && CudaLower &&
```

Comment:
For both this and cudaFree I think it would be easier to always emit both of these and in parallel-lower/cuda lower to fix them up.

Reply:
I will be making a separate pass in a later PR, that moves all of those in a separate cudart-convert-to-gpu-dialect and perhaps a gpu-dialect-to-cpu passes that will better support cuda -> amdcpu conversion as well for example